frontend/ui/wikipedia.lua

```diff
@@ -95,84 +95,44 @@ function Wikipedia:getWikiServer(lang)
     return string.format(self.wiki_server, lang or self.default_lang)
 end
 
--- Say who we are to Wikipedia (see https://meta.wikimedia.org/wiki/User-Agent_policy)
-local USER_AGENT = T("KOReader/%1 (https://koreader.rocks/) %2",
-    (lfs.attributes("git-rev", "mode") == "file" and io.open("git-rev", "r"):read("*line") or "devel"),
-    require('socket.http').USERAGENT:gsub(" ", "/") )
-
--- Codes that getUrlContent may get from requester.request()
-local TIMEOUT_CODE = "timeout" -- from socket.lua
-local MAXTIME_CODE = "maxtime reached" -- from sink_table_with_maxtime
-
--- Sink that stores into a table, aborting if maxtime has elapsed
-local function sink_table_with_maxtime(t, maxtime)
-    -- Start counting as soon as this sink is created
-    local start_secs, start_usecs = ffiutil.gettime()
```

Comment:
No more need for all that/my hard work ? :)
`conn.sock:settimeout(socketutil.total_timeout, 't')` does it all nicely ? Have you really tested it ?

Reply:
It appears to, yeah ;).

Reply:
I mean, it's hard to truly test because my Wi-Fi has decided to be on fire today, but, if I force tiny (1, 2) timeouts, I get the text content just fine, and the logs show a timeout getting the rest of the images:

```
03/11/21-18:37:41 WARN  error getting more images frontend/ui/wikipedia.lua:221: wantread
```

Reply:
Ah, managed to trip earlier, which means I do get nothing and the "Failed to retrieve Wikipedia article" UI feedback, unlike the image case ;).

```
03/11/21-18:39:49 DEBUG error: frontend/ui/wikipedia.lua:221: wantread
```